maintenance/wikia/WikiFactoryVariables/migrateWikiWordmarks.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+* Maintenance script to manage WikiFactory variable for one wiki or multiple wikis (from file)
+* - enable/disable the extension (set to true/false)
+* - set value to the variable (only string, integer, boolean types)
+* - remove variable from the wiki
+* - get the variable value
+* This is one time use script
+* @author Saipetch Kongkatong
+*/
+
+ini_set( 'display_errors', 'stderr' );
+ini_set( 'error_reporting', E_NOTICE );
+
+require_once( dirname( __FILE__ ) . '/../../Maintenance.php' );
+
+use \Wikia\Logger\WikiaLogger;
+
+/**
+ * Class MigrateWikiWordmarks
+ */
+class MigrateWikiWordmarks extends Maintenance {
+
+	protected $dryRun  = false;
+	protected $verbose = false;
+	protected $keyName = '';
+	protected $action = '';
+	protected $success = 0;
+	const WIKI_FACTORY_VARIABLE = "wgOasisThemeSettings";
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Migrates variables in wgOasisThemeSettings to HTTPS";
+		$this->addOption( 'dry-run', 'Dry run mode', false, false, 'd' );
+		$this->addOption( 'verbose', 'Show extra debugging output', false, false, 'v' );
+		$this->addOption( 'keyName', 'Key in WikiFactory variable which should be migrated to https', true, true, 'k' );
+		$this->addOption( 'wikiId', 'Wiki Id', false, true, 'i' );
+		$this->addOption( 'file', 'File of wiki ids', false, true, 'f' );
+		$this->addOption( 'reason', 'Reason to provide when setting a variable', false, true );
+	}
+
+	public function execute() {
+		$this->dryRun  = $this->hasOption( 'dry-run' );
+		$this->verbose = $this->hasOption( 'verbose' );
+		$this->keyName = $this->getOption( 'keyName', '' );
+		$wikiId        = $this->getOption( 'wikiId', '' );
+		$file          = $this->getOption( 'file', '');
+		$reason        = $this->getOption( 'reason' );
+
+		if ( empty( $this->keyName ) ) {
+			die( "Error: Empty key name.\n" );
+		}
+
+		if ( !empty( $wikiId ) ) {
+			$wikiIds = explode(',', $wikiId);
+		} else if ( !empty( $file ) ) {
+			$wikiIds = file( $file );
+		} else {
+			die( "Error: wiki id is empty or the file is invalid.\n" );
+		}
+
+		$varData = (array) WikiFactory::getVarByName( self::WIKI_FACTORY_VARIABLE, $wikiId, true );
+		if ( empty( $varData['cv_id'] ) ) {
+			die( "Error: " . self::WIKI_FACTORY_VARIABLE . " not found.\n" );
+		}
+
+		echo "Variable: " . self::WIKI_FACTORY_VARIABLE . " (Id: $varData[cv_id])" . PHP_EOL;
+		$this->debug( "Variable data: " . json_encode( $varData ) . "\n" );
+
+		$wg = F::app()->wg;
+		$wg->User = User::newFromName( Wikia::BOT_USER );
+		$wg->User->load();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)